components/server/src/billing/entitlement-service.ts

```diff
@@ -104,9 +115,24 @@ export class EntitlementServiceImpl implements EntitlementService {
     ): Promise<MayStartWorkspaceResult> {
         try {
             const billingMode = await this.billingModes.getBillingMode(user.id, organizationId);
+            const organizationSettings = await this.organizationService().getSettings(user.id, organizationId);
+
             switch (billingMode.mode) {
                 case "none":
-                    // if payment is not enabled users can start as many parallel workspaces as they want
+                    // the default limit is MAX_PARALLEL_WORKSPACES_PAID, but organizations can set their own different limit
+                    // we use || here because the default value is 0 and we want to use the default limit if the organization limit is not set
+                    const maxParallelRunningWorkspaces =
+                        organizationSettings.maxParallelRunningWorkspaces || MAX_PARALLEL_WORKSPACES_PAID;
+                    const current = (await runningInstances).filter((i) => i.status.phase !== "preparing").length;
```

Comment:
What's the motivation to exclude "preparing" here? :thinking:

Reply:
Extracted this into an external function in [`1abd308` (#20448)](https://github.com/gitpod-io/gitpod/pull/20448/commits/1abd308507ac567e74107e1a14a1e6d30bef9e2b), like we discussed. Fitting it into the `EntitlementServiceUBP` class didn't feel right, so it's outside of it.